srml/balances/src/lib.rs

```diff
@@ -14,16 +14,131 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Balances: Handles setting and retrieval of free balance,
-//! retrieving total, reserved, and unreserved balances,
-//! repatriating a reserved balance to a beneficiary account that exists,
-//! transfering a balance between accounts (when not reserved),
-//! slashing an account balance, account removal, rewards,
-//! lookup of an index to reclaim an account (when balance not reserved),
-//! increasing total stake.
+//! # Balances Module
 //!
-//! Implements functions for `Currency`, `LockableCurrency`, `TransferAsset`,
-//! and `IsDeadAccount` traits.
+//! ## Overview
+//!
+//! The balances module provides functions for:
+//!
+//! - Getting and setting free balance
+//! - Retrieving total, reserved, and unreserved balances
+//! - Repatriating a reserved balance to a beneficiary account that exists
+//! - Transfering a balance between accounts (when not reserved)
+//! - Slashing an account balance
+//! - Account removal
+//! - Lookup of an index to reclaim an account
+//! - Increasing or decreasing total stake
+//! - Setting and removing locks on chains that implement `LockableCurrency`
+//!
+//! ### Terminology
+//!
+//! - **Existential Deposit:** The existential deposit is the minimum balance required to create or keep an account open. This prevents "dust accounts" from filling storage.
+//! - **Stake:** The total amount of tokens in existence in a system.
+//! - **Reaping an account:** The act of removing an account by resetting its nonce, setting its balance to zero, and removing from storage.
+//! - **Free Balance:** The free balance is the only balance that matters for most operations. When this balance falls below the existential deposit, the account is reaped.
+//! - **Reserved Balance:** Reserved balance still belongs to the account holder, but is suspended. Reserved balance can still be slashed, but only after all of free balance has been slashed. If the reserved balance falls below the existential deposit then it will be deleted.
+//! - **Locks:** Locks enable the runtime to lock an account's balance until a specified block number. Only runtimes that implement the `LockableCurrency` trait allow this.
+//!
+//! ## Interface
+//!
+//! ### Types
+//!
+//! - Balance
+//! - OnFreeBalanceZero
+//! - OnNewAccount
+//! - Event
+//!
+//! These are [associated types](https://doc.rust-lang.org/book/ch19-03-advanced-traits.html#specifying-placeholder-types-in-trait-definitions-with-associated-types) and must be implemented in your `runtime/src/lib.rs`. For example:
+//!
+//! ```ignore
```

Comment:
you can still do "rust, ignore" after the first three dashes to enable rust syntax highlighting without treating it as a test (as shown in [staking docs](https://github.com/paritytech/substrate/pull/1951/files#L89))